### PR TITLE
Fix initial value check for action form config

### PR DESCRIPTION
**Pull request by hetunandu**
fix the initialValue config exists check because `false` value would not set the correct initial value in the action

Also removed some unused Action API methods and renamed others

### CODE REVIEW

File: app/client/src/components/formControls/utils.ts [hunk 1]
```diff
@@ -104,7 +104,7 @@ export const getConfigInitialValues = (config: Record<string, any>[]) => {
         return parseConfig(subSection);
       }
 
-      if (subSection.initialValue) {
+      if ("initialValue" in subSection) {
```
Comment (rishabhsaxena) [inline app/client/src/components/formControls/utils.ts:107]: Just out of curiosity, any advantage of `key in obj` vs `obj.key`?

**Review by rishabhsaxena — APPROVED** [review]

**Review by ApekshaBhosale — APPROVED** [review]

**Comment by hetunandu** [comment]
Merging this in because tests are passing on cypress even if Github actions thinks it is not

**Pull request merged**